openeo/extra/job_management.py

```diff
@@ -0,0 +1,290 @@
+                                _log.info(
+                                    f"Starting job on backend {backend_name} for {row.to_dict()}"
+                                )
+                                job = start_job(
+                                    row=row,
+                                    connection_provider=self.backends[
+                                        backend_name
+                                    ].get_connection,
+                                    connection=self.backends[
+                                        backend_name
+                                    ].get_connection(),
+                                    provider=backend_name,
+                                )
+                            except requests.exceptions.ConnectionError as e:
+                                _log.warning(
+                                    f"Failed to start job for {row.to_dict()}",
+                                    exc_info=True,
+                                )
+                                df.loc[i, "status"] = "start_failed"
+                            else:
+                                df.loc[
+                                    i, "start_time"
+                                ] = datetime.datetime.now().isoformat()
+                                if job:
+                                    df.loc[i, "id"] = job.job_id
+                                    with ignore_connection_errors(context="get status"):
+                                        status = job.status()
+                                        df.loc[i, "status"] = status
+                                        if status == "created":
+                                            # start job if not yet done by callback
+                                            try:
+                                                job.start_job()
+                                                df.loc[i, "status"] = job.status()
+                                            except OpenEoApiError as e:
+                                                _log.error(e)
+                                                df.loc[i, "status"] = "start_failed"
+                                else:
+                                    df.loc[i, "status"] = "skipped"
+
+                            persists(df, output_file)
+
+            time.sleep(self.poll_sleep)
+
+    def on_job_done(self, job: BatchJob, row):
+        """
+        Handles jobs that have finished. Can be overridden to provide custom behaviour.
+
+        Default implementation downloads the results into a folder containing the title.
+
+        :param job: The job that has finished.
+        :param row: DataFrame row containing the job's metadata.
+        """
+        # TODO: param `row` is never accessed in this method. Remove it? Is this intended for future use?
+
+        job_metadata = job.describe_job()
+        job.get_results().download_files(target=job_metadata["title"])
```

Comment:
couple of things:
- This downloads to a folder directly under working dir. I think it's better to have an option to set a different root dir, e.g. through argument in `MultiBackendJobManager.__init__` (defaulting to `"."`)
- It downloads to subfolder based on title, but this creates high risk on overwriting results (if user is not careful about job titles). I'd add a prefix, e.g. using the first 8 or 10 chars of the job id

Reply:
- Added the option for the root directory.
- Avoid jobs overwriting each other's data: In the end it was easier to use the only the job_id in the filenames, instead of a combination of job_id and title.
- To keep things more manageable it makes a subfolder per job, because there can also be result files which are downloaded when the job finishes.